tests/http/testenv/httpd.py

```diff
@@ -104,13 +104,21 @@ def exists(self):
         return os.path.exists(self._cmd)
 
     def set_extra_config(self, domain: str, lines: Optional[Union[str, List[str]]]):
+        if domain not in self._extra_configs:
```

Comment:
As in #14589, the `set_extra` is really mean to replace, not append. Besides purity of naming, remember that test cases *should* be able to run standalone and not rely on settings from previously run tests.